### PR TITLE
fix: Mysql should support more data types. #923

### PR DESCRIPTION
**Pull request by beiwei30**
# Description

fix: Mysql should support more data types. #923

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #923

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: bindings/mysql/mysql.go [hunk 1]
```diff
@@ -285,84 +287,54 @@ func jsonify(rows *sql.Rows) ([]byte, error) {
...
+func convert(columnTypes []*sql.ColumnType, values []interface{}) map[string]interface{} {
+	r := map[string]interface{}{}
 
-			continue
+	for i, ct := range columnTypes {
+		value := values[i]
+
+		switch v := values[i].(type) {
+		case driver.Valuer:
+			if vv, err := v.Value(); err == nil {
```
Comment (pkedy) [inline bindings/mysql/mysql.go:325]: Should `err` be handled?   Feels like it should.

File: bindings/mysql/mysql.go [hunk 2]
```diff
@@ -285,84 +287,54 @@ func jsonify(rows *sql.Rows) ([]byte, error) {
...
+				value = vv
```
Comment (pkedy) [inline bindings/mysql/mysql.go:326]: I'm unsure of this but I'm wondering if `driver.Value` should be converted back to `interface{}`. It's a minor detail but could matter for reflection.

```golang
value = interface{}(vv)
```

File: bindings/mysql/mysql.go [hunk 3]
```diff
@@ -285,84 +287,54 @@ func jsonify(rows *sql.Rows) ([]byte, error) {
...
-		if s, ok := (scanArgs[i]).(*sql.NullTime); ok {
-			r[v.Name()] = s.Time
+func convert(columnTypes []*sql.ColumnType, values []interface{}) map[string]interface{} {
+	r := map[string]interface{}{}
 
-			continue
+	for i, ct := range columnTypes {
+		value := values[i]
+
+		switch v := values[i].(type) {
```
Comment (pkedy) [inline bindings/mysql/mysql.go:323]: How are null values handled? Is there a case needed for those to explicitly set the key to `nil`?

**Review by pkedy — COMMENTED** [review]
Couple questions

File: bindings/mysql/mysql.go (hunk 2 above)
Comment (beiwei30) [reply bindings/mysql/mysql.go]: fixed.

File: bindings/mysql/mysql.go (hunk 1 above)
Comment (beiwei30) [reply bindings/mysql/mysql.go:325]: fixed.

File: bindings/mysql/mysql.go (hunk 3 above)
Comment (beiwei30) [reply bindings/mysql/mysql.go:323]: I revised it to 

```go
		if value != nil {
			r[ct.Name()] = value
		}
```

**Comment by beiwei30** [comment]
@pkedy Phil, would you pls. to review this change again?

**Comment by beiwei30** [comment]
@pkedy any update on this?

**Review by yaron2 — APPROVED** [review]
lgtm

**Pull request merged**